src/CRUD.js

```diff
@@ -1,335 +1,301 @@
+      filtered.push(this.conf.driver.getItem(key) || false);
+    }
 
-/**
- * Returning all entries from database
- * @returns {Array} - collection of entries
- */
-Database.prototype.findAll = function () {
-  return this.select(this.data);
-};
+    if (filtered.length === 0) {
+      collection = this.data;
+    } else if (filtered.length === 1) {
+      collection = filtered[0];
+    } else {
+      collection = intersect.apply(this, filtered);
+    }
 
-/**
- * Dissociate indexed from unindexed keys from an object
- * @param   {Object} obj - object to parse
- * @returns {Array}      - array of indexed keys and unindexed keys
- */
-Database.prototype.getKeys = function ( obj ) {
-  var index, keys = [];
-  keys[0] = {}; // indexed properties
-  keys[1] = {}; // unindexed properties
-
-  for(var property in obj) {
-    index = this.conf.indexedKeys.indexOf(property) !== -1 ? 0 : 1;
-    keys[index][property] = obj[property];
+    // Filtering by unindexed keys
+    return this.filter(this.select(collection), keys.unindexed);
   }
 
-  return keys;
-};
+  /**
+   * Returning all entries from database
+   * @returns {Array} - collection of entries
+   */
+  findAll() {
+    return this.select(this.data);
+  }
 
-/**
- * Retrieve entries from unique keys
- * @param   {Array} collection - array of unique keys
- * @returns {Array}            - array of entries
- */
-Database.prototype.select = function ( collection ) {
-  var data = [];
-  collection = collection.length === 1 ? [collection] : collection;
+  /**
+   * Dissociate indexed from unindexed keys from an object
+   * @param   {Object} obj - object to parse
+   * @returns {Array}      - array of indexed keys and unindexed keys
+   */
+  getKeys( obj ) {
+    let keys = {
+      indexed: {},
+      unindexed: {}
+    };
+
+    for (let property in obj) {
+      let stack = this.conf.indexedKeys.indexOf(property) !== -1
+          ? 'indexed'
+          : 'unindexed';
+
+      keys[stack][property] = obj[property];
+    }
 
-  for (var i = 0, len = collection.length; i < len; i++) {
-    data.push(this.conf.driver.getItem(collection[i]));
+    return keys;
   }
 
-  return data;
-};
-
-/**
- * Filtering a collection of entries based on unindexed keys
- * @private
- * @param   {Array}  collection    - array of entries to search for
- * @param   {Object} unindexedKeys - object of unindexed keys
- * @returns {Array}                - array of entries
- */
-Database.prototype.filter = function ( collection, unindexedKeys ) {
-  var okay, entry, data = [];
+  /**
+   * Retrieve entries from unique keys
+   * @param   {Array} collection - array of unique keys
+   * @returns {Array}            - array of entries
+   */
+  select( collection ) {
+    collection = !Array.isArray(collection) ? [collection] : collection;
 
-  for (var i = 0, len = collection.length; i < len; i++) {
-    entry = collection[i];
-    okay = true;
+    return collection.map((item) => this.conf.driver.getItem(item));
```

Comment:
`.map(this.conf.driver.getItem.bind(this.conf.driver))`

Bonus point for using plain functions instead of prototypes so you don't have to bind...